dashboard/modules/job/job_manager.py

```diff
@@ -206,13 +206,22 @@ async def run(
                 "runtime_env": self._runtime_env,
                 "metadata": self._metadata,
             })
-            # Set RAY_ADDRESS to local Ray address, if it is not set.
+            # Always set RAY_ADDRESS as find_bootstrap_address address for
+            # job submission. In case of local development, prevent user from
+            # re-using http://{address}:{dashboard_port} to interact with
+            # jobs SDK.
+            # TODO:(mwtian) Check why "auto" does not work in entrypoint script
             os.environ[
                 ray_constants.RAY_ADDRESS_ENVIRONMENT_VARIABLE] = \
-                ray._private.services.get_ray_address_from_environment()
+                ray._private.services.find_bootstrap_address().pop()
```

Comment:
Just for my education, what's the bootstrap address?

Reply:
This is a term @mwtian suggested, I also named it "redis" on first revision, then learned there's ongoing migration effort to move redis out of ray critical path, and "bootstrap" covers both.